pkg/controllers/provisioning/provisioner.go

```diff
@@ -343,7 +343,11 @@ func (p *Provisioner) Create(ctx context.Context, n *scheduler.NodeClaim, opts .
 	if err := latest.Spec.Limits.ExceededBy(latest.Status.Resources); err != nil {
 		return "", err
 	}
-	nodeClaim := n.ToNodeClaim(latest)
+	nodeClaim, err := n.ToNodeClaim(latest)
```

Comment:
I'll let it be known that I'm not a huge fan of this call being error-able, but I'm willing to take incremental improvement here if need-be

Reply:
I have now restructured the code to proactively identify this instead of identifying it here in the NodeClaim creation. It is not as good as introducing this into the scheduler but very close. The catch is we will introduce an additional sorting time complexity into the scheduler path.